packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx

```diff
@@ -63,31 +75,176 @@ const useStyles = makeStyles(({ palette, spacing, typography }) => ({
   checkIcon: {
     width: spacing(6),
     height: spacing(6)
+  },
+  collectionName: {
+    color: palette.secondary
   }
 }))
 
-const DetailsTileUnlockedSection = ({
-  premiumConditions
-}: {
-  premiumConditions: PremiumConditions
-}) => {
+type HasAccessProps = {
+  renderDescription: () => ReactNode
+  nftCollection?: Nullable<
+    PremiumConditionsEthNFTCollection | PremiumConditionsSolNFTCollection
+  >
+  followee?: Nullable<User>
+  tippedUser?: Nullable<User>
+}
+
+const DetailsTileUnlockedSection = ({ renderDescription }: HasAccessProps) => {
   const styles = useStyles()
   const neutral = useColor('neutral')
-  const { followee, tippedUser } = useSpecialAccessEntity(premiumConditions)
-  console.log('premiumConditions is', premiumConditions, followee, tippedUser)
 
-  const renderDescription = useCallback(() => {
-    if (premiumConditions.nft_collection) {
+  return (
+    <View style={styles.root}>
+      <View style={styles.headerContainer}>
+        <View style={styles.titleContainer}>
+          <IconUnlocked fill={neutral} />
+          <Text style={styles.title}>{messages.unlocked}</Text>
+        </View>
+        <IconVerifiedGreen style={styles.checkIcon} />
+      </View>
+      {renderDescription()}
+    </View>
+  )
+}
+
+const DetailsTileOwnerSection = ({
+  renderDescription,
+  nftCollection,
+  followee,
+  tippedUser
+}: HasAccessProps) => {
+  const styles = useStyles()
+  const neutral = useColor('neutral')
+
+  return (
+    <View style={styles.root}>
+      <View style={styles.titleContainer}>
+        {nftCollection && (
+          <IconCollectible fill={neutral} width={16} height={16} />
+        )}
+        {(followee || tippedUser) && (
+          <IconSpecialAccess fill={neutral} width={16} height={16} />
+        )}
+        <Text style={styles.title}>
+          {nftCollection ? messages.collectibleGated : messages.specialAccess}
+        </Text>
+      </View>
+      {renderDescription()}
+    </View>
+  )
+}
+
+type DetailsTilePremiumAccessProps = {
+  premiumConditions: PremiumConditions
+  isOwner: boolean
+}
+
+export const DetailsTileHasAccess = ({
+  premiumConditions,
+  isOwner
+}: DetailsTilePremiumAccessProps) => {
+  const styles = useStyles()
+
+  const { nftCollection, followee, tippedUser } =
+    useSpecialAccessEntity(premiumConditions)
+
+  const collectionLink = useMemo(() => {
+    if (!nftCollection) return ''
+
+    const { chain, address, externalLink } = nftCollection
+    if (chain === Chain.Eth && 'slug' in nftCollection!) {
```

Comment:
Hm why do we need the ! here if we already have the null check above?

Reply:
good q. we shouldn't need